packages/eslint-plugin-pf-codemods/src/rules/v6/buttonMoveIconsIconProp/button-moveIcons-icon-prop.ts

```diff
@@ -31,41 +34,60 @@ module.exports = {
             ) {
               const variantProp = getAttribute(node.openingElement, "variant");
               const iconProp = getAttribute(node.openingElement, "icon");
-              if (!variantProp || iconProp) {
+              if (iconProp) {
                 return;
               }
               const variantValue = getAttributeValue(
                 context,
-                variantProp.value
+                variantProp?.value
               );
 
               const isEnumValuePlain =
                 buttonVariantEnumImport &&
                 variantValue?.object?.name ===
                   buttonVariantEnumImport.local.name &&
                 variantValue?.property.name === "plain";
-              if (variantValue !== "plain" && !isEnumValuePlain) {
-                return;
-              }
+
+              const isPlain = variantValue === "plain" || isEnumValuePlain;
+
               const childrenProp = getAttribute(node, "children");
-              let childrenValue;
+
+              let childrenValue: string | undefined;
               if (childrenProp) {
                 const childrenPropExpression = getExpression(
                   childrenProp?.value
                 );
                 childrenValue = childrenPropExpression
-                  ? context.getSourceCode().getText(childrenPropExpression)
+                  ? source.getText(childrenPropExpression)
                   : "";
-              } else {
+              } else if (isPlain) {
                 childrenValue = getChildrenAsAttributeValueText(
                   context,
                   node.children
                 );
               }
-              if (!childrenValue) {
+
+              if (!childrenValue && isPlain) {
+                return;
+              }
+
+              const iconComponentChild = getChildElementByName(node, "Icon");
```

Comment:
I think we should check an `import { Icon }` from react-core, otherwise this will get any element named "Icon".